core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -867,6 +867,9 @@ public PlanOptimizers(
                 ImmutableSet.of(
                         new AddIntermediateAggregations(),
                         new RemoveRedundantIdentityProjections())));
+        // Run MetadataQueryOptimizer again to optimize partial aggregations
+        builder.add(new MetadataQueryOptimizer(metadata));
```

Comment:
Can this invalidate remote or local exchanges picked above?

Also, MetadataQueryOptimizer doesn't distinguish between partial, intermediate and final aggregations, which may result in an incorrect plan, especially for aggregate functions other than min, max (where single step = final step). For example, does this work for `approx_distinct` supported by this optimizer?